var/spack/repos/builtin/packages/charm/package.py

```diff
@@ -169,4 +170,19 @@ def install(self, spec, prefix):
         # this wouldn't be difficult.
         build = Executable(join_path(".", "build"))
         build(target, version, *options)
+
+        # Charm++'s install script does not copy files, it only creates
+        # symbolic links. Fix this.
+        for dirpath, dirnames, filenames in os.walk(prefix):
+            for filename in filenames:
+                filepath = join_path(dirpath, filename)
+                if os.path.islink(filepath):
+                    tmppath = filepath + ".tmp"
+                    # Skip dangling symbolic links
+                    try:
+                        shutil.copy2(filepath, tmppath)
+                        os.remove(filepath)
+                        os.rename(tmppath, filepath)
```

Comment:
No... Charm++ builds in a subdirectory `tmp` of the install directory. (Yes, I know...) When the build script finishes, the install directory is already populated with symbolic links.

There might be other ways to build, but Charm++'s build mechanism is quite non-standard, and I lack the knowledge to modify this. If it's important, we can inquire with the Charm++ developers.